tests/npc_test.cpp

```diff
@@ -315,75 +316,105 @@ static void check_npc_movement( const tripoint &origin )
     }
 }
 
-TEST_CASE( "npc-board-player-vehicle" )
+static npc *make_companion( const tripoint &npc_pos )
 {
-    // Place the player at the center of the map
-    const tripoint player_pos( 60, 60, 0 );
-    g->place_player( player_pos );
-    // Clear the map so the NPC can move unimpeded and there are no obstacles
-    clear_map();
-
-    map &here = get_map();
-    Character &pc = get_player_character();
-
-    // Place the NPC outside the door and down one
-    tripoint npc_pos = player_pos + point( -2, -1 );
-
-    // Place an ambulance at the player's position, rotated correctly, 100% fuel, perfect condition
-    vehicle *veh = here.add_vehicle( vehicle_prototype_ambulance, player_pos, -90_degrees, 100, 2 );
-    // And make sure the game knows the player is in it so the NPC can follow
-    here.board_vehicle( player_pos, &pc );
-    REQUIRE( veh != nullptr );
-
-    // Create the NPC
     shared_ptr_fast<npc> guy = make_shared_fast<npc>();
     guy->normalize();
     guy->randomize();
-    // Spawn the NPC in the location desired
     guy->spawn_at_precise( tripoint_abs_ms( get_map().getabs( npc_pos ) ) );
     overmap_buffer.insert_npc( guy );
     g->load_npcs();
     guy->companion_mission_role_id.clear();
-    // Ensure they aren't trying to guard anything
     guy->guard_pos = std::nullopt;
-    // Reset the NPC, so they don't need to eat, or want to play with their items
     clear_character( *guy );
-    // But set them back to the desired position, because clear_character also sets position
     guy->setpos( npc_pos );
-
-    // Make the NPC a follower of the player
```

Comment:
I don't think these comments are without value. They're saying why these things are being done because they're specific problems that came up in testing, and I think are thus important things to understand when writing a test using this.
In particularly, I think the setpos comment is important (if I'm reading this code and seeing that we're spawning at a precise location, then setting the position back to that again, I have a question. The comment answers that without spending my time).

Reply:
Okay, for one I'm mostly not a fan of comments in code, so there should be a minimal amount of that. That said especially the `setpos` is pretty unintuitive, so some explanation might be worthwhile. But the reason I extracted that to it's own method is so people using it don't have to think about it and just get an NPC companion at the point passed to it.